subprojects/docs/src/samples/groovy/incrementalCompilation/groovy/build.gradle

```
@@ -0,0 +1,21 @@
+subprojects {
+    apply plugin: 'groovy'
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        implementation localGroovy()
+    }
+
+// tag::enable-groovy-incremental[]
+    tasks.withType(GroovyCompile) {
```

```suggestion
    tasks.withType(GroovyCompile).configureEach {
```

If not, I think `withType(_, Action)` is eager.